services/src/main/java/org/keycloak/services/resources/admin/AdminRoot.java

```diff
@@ -222,6 +242,11 @@ public Object getRealmsAdmin(@Context final HttpHeaders headers) {
      */
     @Path("serverinfo")
     public Object getServerInfo(@Context final HttpHeaders headers) {
+
+        if (!isAdminApiEnabled()) {
```

Comment:
Should we really block server info requests if admin-api is not enabled?

Users might still be interested in getting server info from an instance.

Reply:
I'd leave this as it is now